### PR TITLE
Coverage types refactor and added corpus size in UI

### PR DESCRIPTION
**Pull request by ggrieco-tob**

### CODE REVIEW

File: lib/Echidna/Types/Corpus.hs [hunk 1]
```diff
@@ -1,6 +1,9 @@
 module Echidna.Types.Corpus where
 
-import Data.Set (Set)
+import Data.Set (Set, size)
 import Echidna.Types.Tx (Tx)
 
 type Corpus = Set (Integer, [Tx])
```
Comment (incertia) [inline lib/Echidna/Types/Corpus.hs:6]: consider `newtype`-ing this if you want type-specific size/whatever

File: lib/Echidna/Types/Coverage.hs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+-- | Given good point coverage, count unique points.
+coveragePoints :: CoverageMap -> Int
+coveragePoints = sum . fmap size
+
+-- | Given good point coverage, count the number of unique points but
+-- only considering the different instruction PCs (discarding the TxResult).
+-- This is useful to report a coverage measure to the user
+scoveragePoints :: CoverageMap -> Int
+scoveragePoints = sum . fmap (size . Data.Set.map fst3)
+
```
Comment (incertia) [inline lib/Echidna/Types/Coverage.hs:28]: extra newlines

File: lib/Echidna/Types/Corpus.hs [hunk 3]
```diff
@@ -1,6 +1,9 @@
 module Echidna.Types.Corpus where
 
-import Data.Set (Set)
+import Data.Set (Set, size)
```
Comment (incertia) [inline lib/Echidna/Types/Corpus.hs:3]: `length` is defaulted to `size`

https://hackage.haskell.org/package/containers-0.6.4.1/docs/src/Data.Set.Internal.html#line-317

semantics up to you

**Review by incertia — CHANGES_REQUESTED** [review]

File: lib/Echidna/Types/Coverage.hs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+-- | Given good point coverage, count unique points.
+coveragePoints :: CoverageMap -> Int
+coveragePoints = sum . fmap size
+
+-- | Given good point coverage, count the number of unique points but
+-- only considering the different instruction PCs (discarding the TxResult).
+-- This is useful to report a coverage measure to the user
+scoveragePoints :: CoverageMap -> Int
+scoveragePoints = sum . fmap (size . Data.Set.map fst3)
```
[the remark this answers is not in the document]
Comment (neuromancer) [reply lib/Echidna/Types/Coverage.hs:27]: Done

[the remark this answers is not in the document]
Comment (ggrieco-tob) [reply lib/Echidna/Types/Coverage.hs:27]: Done.

File: lib/Echidna/Types/Coverage.hs (hunk 2 above)
Comment (ggrieco-tob) [reply lib/Echidna/Types/Coverage.hs:28]: Done

**Pull request merged**